chore(core): Update API Design Guide (#8155)
island-is/island.is: handbook/technical-overview/api-design-guide/security.md

# Security

## OWASP Top 10

> Thw OWASP Foundation is known for its top ten web application security risks.
> Now the foundation also collected the top ten security considerations for API projects.
> 
> A developer of an API should know of and review these items regularly against its APIs.

Reviewer: ```suggestion
The OWASP Foundation is known for its top ten web application security risks.
Now the foundation has also collected the top ten security considerations for API projects.

Developers should know of and review these items regularly against their APIs.
```
